File: modinverse.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int x;
    cin >> x;
    int m;
    cin >> m;
    int count = 0;
    for (int n = 0; n<=m; n++){
        int prod = x*n;
        if(prod%m == 1){
            cout << n<<"\n";
            count ++;
            break;
        }
    
    }
    if (count == 0){
      cout << "No such integer exists.";
    }
    
    return 0;
}

File: Python Solutions/modinverse.py
#debug
x = int(input())
m = int(input())
for n in range(1, m+1):
    if (x*n)%m == 1 and m%n == x:
        print(n)
        break

if (x*n)%m != 1 and m%n != x:
    print("No such integer exists")
        